### PR TITLE
#1517 new ctor for ResourceOf

### PR DESCRIPTION
**Pull request by yegor256**
I added the ctor, as per #1517

### CODE REVIEW

File: src/test/java/org/cactoos/io/ResourceOfTest.java [hunk 1]
```diff
@@ -75,6 +75,20 @@ public void readsTextResource() throws Exception {
...
+    public void readsTextResourceThroughClassloader() throws Exception {
+        new Assertion<>(
+            "Can't read a text resource from classloader",
+            new TextOf(
+                new ResourceOf(
+                    "org/cactoos/large-text.txt",
+                    ResourceOfTest.class
+                )
+            ).asString(),
+            Matchers.endsWith(" laborum.\n")
```
Comment (andreoss) [inline src/test/java/org/cactoos/io/ResourceOfTest.java:88]: @yegor256 There's `EndsWith` matcher available for `Text`, no need to call `.asString()` explicitly.

**Review by andreoss — CHANGES_REQUESTED** [review]

**Comment by victornoel** [comment]
@yegor256 the build does not pass and REV left some remarks

**Comment by yegor256** [comment]
@victornoel fixed

**Review by andreoss — APPROVED** [review]

File: src/test/java/org/cactoos/io/ResourceOfTest.java (hunk 1 above)
Comment (victornoel) [reply src/test/java/org/cactoos/io/ResourceOfTest.java:90]: @yegor256 can you fix this too?

File: src/test/java/org/cactoos/io/ResourceOfTest.java [hunk 2]
```diff
@@ -63,15 +64,30 @@ public void readsBinaryResource() throws Exception {
...
                 )
+            ),
+            new EndsWith("est laborum.\n")
+        ).affirm();
+    }
+
+    @Test
+    public void readsTextResourceThroughClassloader() throws Exception {
+        new Assertion<>(
+            "Can't read a text resource from classloader",
```
Comment (victornoel) [inline src/test/java/org/cactoos/io/ResourceOfTest.java:83]: @yegor256 in cactoos we favour the positive form in test messages "Must..."

File: src/test/java/org/cactoos/io/ResourceOfTest.java [hunk 3]
```diff
@@ -63,15 +64,30 @@ public void readsBinaryResource() throws Exception {
     }
 
     @Test
+    @SuppressWarnings("PMD.AvoidDuplicateLiterals")
```
Comment (victornoel) [inline src/test/java/org/cactoos/io/ResourceOfTest.java:67]: @yegor256 why do we need this at the method level? if it's needed let's just use it once at class level for consistency

**Review by victornoel — CHANGES_REQUESTED** [review]
@yegor256 a few more comments

**Comment by yegor256** [comment]
@victornoel fixed in bc60ff88

**Comment by victornoel** [comment]
@rultor merge

**Comment by victornoel** [comment]
@yegor256 thx

**Comment by 0crat** [comment]
Job [#1520](https://github.com/yegor256/cactoos/issues/1520) is not in the agenda of @andreoss[/z](https://www.0crat.com/u/andreoss), can't set title

**Comment by 0crat** [comment]
Job [#1520](https://github.com/yegor256/cactoos/issues/1520) is not in the agenda of @andreoss[/z](https://www.0crat.com/u/andreoss), can't set title

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @andreoss[/z](https://www.0crat.com/u/andreoss), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Code review was too long (13 days), architects (@victornoel) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by sereshqua** [comment]
@andreoss please make sure you will find at least 3 issues during next CR, thanks

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request merged**